pytorch_lightning/trainer/trainer.py

```diff
@@ -1479,16 +1479,15 @@ def _call_callback_hooks(
         hook_name: str,
         *args: Any,
         **kwargs: Any,
-    ) -> Optional[Any]:
-        output = None
+    ) -> None:
```

Comment:
why are we returning None for callback hooks? I thought part of the motivation for https://github.com/PyTorchLightning/pytorch-lightning/issues/8506 was to allow returning something for callback hooks? i.e. #4 in the description

Reply:
4) was described as a limitation, but it's not something we want to support with the refactor. However, the refactor allows for adding it easily when we want to.